### PR TITLE
[Feedback Wanted] Proposal for supporting ARM

### PR DESCRIPTION
**Pull request by tesych**
We wanted to introduce the proposals to our repository to enable our customers to participate even more in planning of how new features will be implemented.

### CODE REVIEW

**Comment by Greendogo** [comment]
Does "NVIDIA Jetson Nano or similar embedded dev platforms" imply the NVIDIA Jetson family (TK1, TX1/2, Nano and Xavier).  For example, Xavier CPU is 8-core Carmel ARM v8.2 64-bit, while Nano is Quad-core ARM Cortex-A57.

Also, this family of devices has built in NVIDIA GPU capabilities and currently the Kinect SDK uses GPU processing for some things.  Will these built-in GPU capabilities be supported?

Is this ARM support going to go beyond the NVIDIA Jetson family of computers?

File: proposals/ARM-support.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    - CPU: Quad-Core ARM Cortex-A57 @ 1.43 GHz
+    - 4GB DDR4 RAM
+- Jetson TX2 (to run Body Tracking SDK)
+    - GPU: Custom 256 CUDA Core GPU (Pascal architecture)
+    - CPU: Dual-Core NVIDIA Denver 2 ARMv8 64-bit processor + Quad-Core ARM Cortex-A57 @ 1.43 GHz
+    - 4GB / 8GB DDR4 RAM
+
+## Implementation
+
+There are several large adjustments need to be implemented: //WES: there are 4 repo's that need to be modified, SDK, BT, Depth, and packaging.
```
Comment (wes-b) [inline proposals/ARM-support.md]:
```suggestion
There are several large adjustments need to be implemented:
```

File: proposals/ARM-support.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+- Changes to Sensor SDK, Depth Engine, Body Tracking SDK and packaging repo's
+- Changes to the build process for Sensor SDK, Depth Engine and Body Tracking SDK
+- Changes to the test system for Sensor SDK, Depth Engine and Body Tracking SDK
+- Changes to the release process for both Linux and Windows.
+- Evaluate and make changes if needed to the depth engine code that uses SSE instructions.
+
+We will target releasing ARM binaries for Windows and Linux.
```
Comment (wes-b) [inline proposals/ARM-support.md]: I think we should pick 1, Windows or Linux. Any data on which one?

File: proposals/ARM-support.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+## Packaging
+
+### Sensor SDK
+
+- Microsoft installer (MSI)
+- Nuget package
```
Comment (wes-b) [inline proposals/ARM-support.md]:
```suggestion
- Nuget package (for Windows)
```

File: proposals/ARM-support.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+- Deb package
```
Comment (wes-b) [inline proposals/ARM-support.md]:
```suggestion
- Deb package (For Ubuntu 18.04)
```

File: proposals/ARM-support.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+### Body Tracking SDK
+
+- Microsoft installer (MSI)
+- Nuget package
```
Comment (wes-b) [inline proposals/ARM-support.md]:
```suggestion
- Nuget package (For Windows)
```

File: proposals/ARM-support.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+- Deb package
```
Comment (wes-b) [inline proposals/ARM-support.md]:
```suggestion
- Deb package (For Ubuntu 18.04)
```

File: proposals/ARM-support.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
+
+# Support for ARM
+
+* [x] Proposed 10/31/2019
```
Comment (wes-b) [inline proposals/ARM-support.md]:
```suggestion
## Proposal State

* [x] Proposed 10/31/2019
```

File: proposals/ARM-support.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+* [ ] Prototype: Not Started
+* [ ] Implementation: Not Started
+* [ ] Specification: Not Started
+
+## Summary
+
+Support the Sensor and Body Tracking SDKs on ARM based boards.
```
Comment (wes-b) [inline proposals/ARM-support.md]: Suggestion: Move this section before the proposal state

File: proposals/README.md [hunk 9]
```diff
@@ -0,0 +1,6 @@
+# Azure Kinect Sensor SDK Proposals
+
+New features proposals are living documents describing the current thinking about a given feature. Proposals are the way for Azure Kinect team  to communicate to the public upcoming new feature work and get the feedback.
```
Comment (wes-b) [inline proposals/README.md]: Can we add a couple sections here? 
* One about the stages or the proposal process from initial idea to completed feature.
* One about where users should direct feedback (the proposal up until the feature is checked in)

File: proposals/README.md [hunk 10]
```diff
@@ -0,0 +1,6 @@
...
+At this time, all proposals will be created by Microsoft Azure Kinect team for the most desired features, but if you have a great idea for a proposal, please submit it on [User Voice](https://feedback.azure.com/forums/920053).
```
Comment (wes-b) [inline proposals/README.md]: When will MS kick off a new proposal?
How much time do we give everyone to review?

File: proposals/README.md [hunk 11]
```diff
@@ -0,0 +1,6 @@
...
+
+We would love your feedback on every proposal and therefore we will create a GitHub issue where you can prove it.
```
Comment (wes-b) [inline proposals/README.md]: Not sure what we are tying to say here

Comment (tesych) [reply proposals/README.md]: @Greendogo, we picked Nvidia Jetson family devices for a few reasons:

- It is something that most of the users were asking on Customer voice

- it fits Azure Kinect DK both SDK's requirements including GPU usage, so the built in GPU capabilities will be supported.


For this proposal, we will focus on those 2 boards only and we will consider other types of devices if we see the demand for it.

**Comment by tesych** [comment]
@Greendogo , we picked Nvidia Jetson family devices for a few reasons:
It is something that most of the users were asking on Customer voice
it fits Azure Kinect DK both SDK's requirements including GPU usage, so the built in GPU capabilities will be supported.
For this proposal, we will focus on those 2 boards only and we will consider other types of devices if we see the demand for it.

**Comment by nonlinear1** [comment]
Add  Nvidia Jetson Xavier board support on Ubuntu 18.04. the nano and jetson tx2 are lower computing ability than Xavier. For me and many other customers bought kinect for azure for developing slam. The two boards are hard to run slam for its lower computing ability.
Thanks a lot

File: proposals/ARM-support.md (hunk 2 above)
Comment (sjdrc) [reply proposals/ARM-support.md]: I'd imagine the majority of people wanting to use Azure Kinect with ARM want to do so running linux on an SBC (such as the jetson)

**Comment by sjdrc** [comment]
Is there any sort of rough timeline on progress for ARM support? It is critical for our use case and we'd rather not put energy into building workarounds if ARM support is coming sooner than we anticipate.

File: proposals/ARM-support.md (hunk 2 above)
Comment (tesych) [reply proposals/ARM-support.md]: The request we received was for Linux, so we will focus on Linux implementation for this effort.

**Comment by Greendogo** [comment]
I'm not sure who is making the decision 'not' to support the Xavier board, but it is definitely better for Slam, so I'm not sure why you'd exclude it.  You may want to bring this up with your Product Manager. In so doing, please evaluate the similarities between the Xavier and its predecessor the TX2 - getting Xavier support is probably not an expensive ask on top of the TX2.

**Comment by nonlinear1** [comment]
I also don't understand why they don't support Xavier, tx2 is a outdated product.

File: proposals/README.md (hunk 11 above)
Comment (tesych) [reply proposals/README.md]: fixed it

**Comment by tesych** [comment]
@sjdrc the plan is to complete the work by end of February.

File: proposals/README.md (hunk 9 above)
Comment (tesych) [reply proposals/README.md]: added those changes

**Comment by tesych** [comment]
the review process is completed.

**Comment by compiaffe** [comment]
@tesych Do you have any update on the ARM support implementation?
Is the End of February ETA still correct?

**Comment by wes-b** [comment]
We won't have it by the end of the month but we are continuing to work on it.

**Comment by xunsamama** [comment]
> b

Any update?

**Comment by wes-b** [comment]
Released today

**Pull request merged**